api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ObservationValueTranslatorImpl.java

```diff
@@ -60,6 +58,15 @@ public Type toFhirResource(@Nonnull Obs obs) {
 			if (obs.getConcept() instanceof ConceptNumeric) {
 				ConceptNumeric cn = (ConceptNumeric) obs.getConcept();
 				result.setUnit(cn.getUnits());
+				
+				// only set the coding system if unit conforms to UCUM standard
+				Coding coding = quantityCodingTranslator.toFhirResource(cn);
+				if (coding.hasSystem() && coding.getSystem().equals(UCUM_SYSTEM_URI)) {
```

Comment:
ignoring every system but UCUM. is this the correct approach or do we want to use a fallback system for observation quantity?

Reply:
I think having the hard-coded to UCUM seems like it will work. Most real-world uses probably only care about what's in the `unit` field... The coded representation of the unit is basically useful if you want to use a coding system to translate into another unit... and the only coding system I'm aware of that has those kinds of unit conversion rules is UCUM.